rpc/lib/server/handlers.go

```diff
@@ -109,35 +109,35 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 		var request types.RPCRequest
 		err := json.Unmarshal(b, &request)
 		if err != nil {
-			WriteRPCResponseHTTPError(w, http.StatusBadRequest, types.NewRPCResponse("", nil, fmt.Sprintf("Error unmarshalling request: %v", err.Error())))
+			WriteRPCResponseHTTP(w, types.RPCParseError(""))
```

Comment:
The Error response provides a base message corresponding to the error code.  In my first version of this code, I had included the "data" field in the error that could have been used to add additional info - but I removed it based on earlier feedback.    Logging will probably work just fine.